Fragment management and navigation basic setup

Conyacr opened the pull request:
This is the navigation drawer and fragment replacement system working for the app, android 5.0 or api 21 required for usesage.

Conyacr committed:
Color palette

Testing color scheme

Conyacr committed:
Merged branch master into Ryan

Conyacr committed:
Super convoluted navigation drawer and fragments

I will have a more detailed usage on this thursday for the repo

Conyacr committed:
Super convaluted navigation drawer and fragments

I will have a more detailed usage on this thursday for the repo

Conyacr committed:
Merged branch Ryan into Ryan

Conyacr committed:
Navigation drawer and fragment factory created

This is a working set for the fragment replacement that will be used for the app, it does how ever require the usage of an android 5.0 or api 21 device to run properly

gklundt commented:
**minor/petty:** "Service" package should be "service" as a matter of convention.

**structure/more than petty:**
The view objects really don't belong in the service package because they are not associated with the data model.  I think these are better suited to go in either the root package or in a view package since they are part of the UI and not the service 
- CustomExpandableListAdapter.java
- FragmentNavigationManager.java
- NavigationManager.java

The main reason for this is that the service package should be replaced by an implementation of the service that gets its data from a database rather than a static model at some point.  Right now, if the service package were replaced, we would lose UI functionality.

Conyacr committed:
Fixed a bug with the navigation on click listener

Found an issue with the on click and how it checked which of the fragments to replace the view with
